homeassistant/components/broadlink/remote.py

```diff
@@ -0,0 +1,376 @@
+        self._state = True
+        self._available = True
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the remote."""
+        return self._unique_id
+
+    @property
+    def is_on(self):
+        """Return True if the remote is on."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return True if the remote is available."""
+        return self._available
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_LEARN_COMMAND
+
+    @callback
+    def flags(self):
+        """Return dictionary of toggle flags.
+
+        A toggle flag indicates whether `self._async_send_code()`
+        should send an alternative code for a key device.
+        """
+        return self._flags
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the remote on."""
+        self._state = True
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the remote off."""
+        self._state = False
+
+    async def async_update(self):
+        """Update the availability of the remote."""
+        if not self.available:
+            await self._async_connect()
+
+    async def async_load_storage_files(self):
+        """Load codes and toggle flags from storage files."""
+        try:
+            self._codes.update(
+                await self._async_get_data(self._code_storage, CODE_STORAGE_SCHEMA)
+            )
+            self._flags.update(
+                await self._async_get_data(self._flag_storage, FLAG_STORAGE_SCHEMA)
+            )
+        except (HomeAssistantError, vol.MultipleInvalid):
+            return False
+        return True
+
+    async def _async_get_data(self, storage, schema):
+        """Return data from storage file."""
+        try:
+            data = schema(await storage.async_load() or {})
+        except (HomeAssistantError, vol.MultipleInvalid) as err:
+            _LOGGER.error("Failed to load '%s': %s", storage.path, err)
+            raise
+        return data
+
+    async def async_send_command(self, command, **kwargs):
+        """Send a list of commands to a device."""
+        kwargs[ATTR_COMMAND] = command
+        kwargs = SERVICE_SEND_SCHEMA(kwargs)
+        commands = kwargs[ATTR_COMMAND]
+        device = kwargs[ATTR_DEVICE]
+        repeat = kwargs[ATTR_NUM_REPEATS]
+        delay = kwargs[ATTR_DELAY_SECS]
+
+        if not self._state:
+            return
+
+        should_delay = False
+        for _, cmd in product(range(repeat), commands):
+            try:
+                should_delay = await self._async_send_code(
+                    cmd, device, delay if should_delay else 0
+                )
+            except ConnectionError:
+                break
+
+        self._flag_storage.async_delay_save(self.flags, FLAG_SAVE_DELAY)
```

Comment:
Don't pass `self.flags` but pass a function that returns the data to save.

Reply:
It was a callback function with a terrible name, so I changed it.